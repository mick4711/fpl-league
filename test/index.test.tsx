/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react"
import renderer from "react-test-renderer"
import Table from "../pages/index"
import { validLeagueData, emptyLeagueData } from "./mock-data"

beforeAll(() => {
  jest.useFakeTimers()
  jest.setSystemTime(new Date("09 Apr 1957 19:11:00 GMT").getTime())
})

afterAll(() => {
  jest.useRealTimers()
})

describe("Table component", () => {
  it("renders with valid playerData", () => {
    const { getByRole, getByText } = render(
      <Table leagueData={validLeagueData} apiLeague="testLeague" />
    )

    const title = getByText("Fantasy Football customisable tables", {
      exact: false,
    })
    const main = getByRole("main")
    const table = getByRole("table")

    expect(title).toBeInTheDocument()
    expect(main).toBeInTheDocument()
    expect(table).toBeInTheDocument()
    expect(table.className).toEqual(expect.stringContaining("leaguetable"))
  })

  it("renders with empty playerData and message", () => {
    const { getByRole, getByText } = render(<Table leagueData={emptyLeagueData} apiLeague="testLeague" />)

    const message = getByText("Gameweek: No data returned from server")
    const title = getByText("Fantasy Football customisable tables", {
      exact: false,
    })
    const main = getByRole("main")
    const table = getByRole("table")

    expect(message).toBeInTheDocument()
    expect(title).toBeInTheDocument()
    expect(main).toBeInTheDocument()
    expect(table).toBeInTheDocument()
    expect(table.className).toEqual(expect.stringContaining("leaguetable"))
  })

  it("renders snapshot correctly", () => {
    const component = renderer.create(<Table leagueData={validLeagueData} apiLeague="testLeague" />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
