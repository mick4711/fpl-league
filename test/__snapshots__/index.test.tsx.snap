// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component renders snapshot correctly 1`] = `
<div>
  <main
    className="main"
  >
    <h1
      className="title"
    >
      Next.js
      <br />
      Fantasy Football customisable tables
    </h1>
    <div
      className="table"
    >
      <p>
        Gameweek: 
        10
      </p>
      <p
        suppressHydrationWarning={true}
      >
        Time:
         
        Tue 9 Apr 1957, 19:11
      </p>
      <table
        className="leaguetable"
      >
        <thead>
          <tr
            className="shaded"
          >
            <td>
              Name
            </td>
            <td>
              Team
            </td>
            <td
              className="sort sorted"
              onClick={[Function]}
            >
              Points
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "display": "inline-block",
                    "margin": 0,
                    "maxWidth": "100%",
                    "overflow": "hidden",
                    "position": "relative",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "display": "block",
                      "maxWidth": "100%",
                    }
                  }
                >
                  <img
                    alt=""
                    aria-hidden={true}
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUiIGhlaWdodD0iMTUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style={
                      Object {
                        "border": "none",
                        "display": "block",
                        "margin": 0,
                        "maxWidth": "100%",
                        "padding": 0,
                      }
                    }
                  />
                </div>
                <img
                  alt="down arrow"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "visibility": "hidden",
                      "width": 0,
                    }
                  }
                />
              </div>
            </td>
            <td>
              Rank
            </td>
            <td
              className="sort "
              onClick={[Function]}
            >
              GW Points
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "display": "inline-block",
                    "margin": 0,
                    "maxWidth": "100%",
                    "overflow": "hidden",
                    "position": "relative",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "display": "block",
                      "maxWidth": "100%",
                    }
                  }
                >
                  <img
                    alt=""
                    aria-hidden={true}
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUiIGhlaWdodD0iMTUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style={
                      Object {
                        "border": "none",
                        "display": "block",
                        "margin": 0,
                        "maxWidth": "100%",
                        "padding": 0,
                      }
                    }
                  />
                </div>
                <img
                  alt="down arrow"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "visibility": "hidden",
                      "width": 0,
                    }
                  }
                />
              </div>
            </td>
            <td>
              GW Rank
            </td>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Player 1
            </td>
            <td>
              <a
                href="https://fantasy.premierleague.com/entry/1/event/10"
                rel="noreferrer"
                target="_blank"
              >
                Team 1
              </a>
            </td>
            <td
              className="numeric"
            >
              100
            </td>
            <td
              className="numeric"
            >
              11
            </td>
            <td
              className="numeric"
            >
              25
            </td>
            <td
              className="numeric"
            >
              33
            </td>
          </tr>
          <tr>
            <td>
              Player 2
            </td>
            <td>
              <a
                href="https://fantasy.premierleague.com/entry/2/event/10"
                rel="noreferrer"
                target="_blank"
              >
                Team 2
              </a>
            </td>
            <td
              className="numeric"
            >
              75
            </td>
            <td
              className="numeric"
            >
              22
            </td>
            <td
              className="numeric"
            >
              50
            </td>
            <td
              className="numeric"
            >
              11
            </td>
          </tr>
          <tr>
            <td>
              Player 3
            </td>
            <td>
              <a
                href="https://fantasy.premierleague.com/entry/3/event/10"
                rel="noreferrer"
                target="_blank"
              >
                Team 3
              </a>
            </td>
            <td
              className="numeric"
            >
              50
            </td>
            <td
              className="numeric"
            >
              33
            </td>
            <td
              className="numeric"
            >
              30
            </td>
            <td
              className="numeric"
            >
              22
            </td>
          </tr>
        </tbody>
      </table>
      <div
        className="notes"
      >
        <ul
          className="bullets"
        >
          <li>
            Next.js static rendered: data fetched and added at build time, no data fetched from front-end.
          </li>
          <li>
            Incremental Static Regeneration: page is regenerated if data has updated.
          </li>
          <li>
            Stale page is served while the updated page is being regenerated.
          </li>
          <li>
            Data fetched: 
            2021-04-27 09:26:36.715872
             GMT
          </li>
        </ul>
      </div>
    </div>
  </main>
</div>
`;
